prettyPrinters/gdb_providers.py

```diff
@@ -255,7 +255,7 @@ def __init__(self, valobj):
         # type: (Value) -> None
         # BACKCOMPAT: Rust 1.62.0. Drop `else`-branch
         value = valobj["value"]
-        if value["pointer"]:
+        if value.type.code == gdb.TYPE_CODE_STRUCT:
```

Comment:
Works for me but I'd suggest to make it clearer - check if `pointer` field exists explicitly (using some other API). Otherwise, this code forces reader to think more than I expect